File: java/java-sdk/src/main/java/io/kestra/sdk/model/Scope.java
/*
 * Kestra EE
 * All API operations, except for Superadmin-only endpoints, require a tenant identifier in the HTTP path.<br/> Endpoints designated as Superadmin-only are not tenant-scoped.
 */


package io.kestra.sdk.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Gets or Sets Scope
 */
public enum Scope {
  
  STATIC("STATIC"),
  
  INSTANCE("INSTANCE"),
  
  TENANT("TENANT"),
  
  NAMESPACE("NAMESPACE"),
  
  UNKNOWN_DEFAULT_OPEN_API("unknown_default_open_api");

  private String value;

  Scope(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static Scope fromValue(String value) {
    for (Scope b : Scope.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    return UNKNOWN_DEFAULT_OPEN_API;
  }
}

File: python/python-sdk/kestrapy/models/scope.py
# coding: utf-8

"""
    Kestra EE

    All API operations, except for Superadmin-only endpoints, require a tenant identifier in the HTTP path.<br/> Endpoints designated as Superadmin-only are not tenant-scoped.
"""  # noqa: E501


from __future__ import annotations
import json
from enum import Enum
from typing_extensions import Self


class Scope(str, Enum):
    """
    Scope
    """

    """
    allowed enum values
    """
    STATIC = 'STATIC'
    INSTANCE = 'INSTANCE'
    TENANT = 'TENANT'
    NAMESPACE = 'NAMESPACE'

    @classmethod
    def from_json(cls, json_str: str) -> Self:
        """Create an instance of Scope from a JSON string"""
        return cls(json.loads(json_str))
